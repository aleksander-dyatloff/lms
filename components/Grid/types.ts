namespace GridComponent {
  export interface Props {
  }
}

export default GridComponent
