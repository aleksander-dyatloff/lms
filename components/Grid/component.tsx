import { FC } from 'react'

import Wrapper from './styles'
import GridComponent from './types'

const Grid: FC<GridComponent.Props> = ({ ...restProps }) => (
  <Wrapper
    {...restProps}
  />
)

export default Grid
