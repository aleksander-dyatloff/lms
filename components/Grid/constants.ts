export const gridColumnsCount = 12

export const columnSize = 100 / gridColumnsCount
