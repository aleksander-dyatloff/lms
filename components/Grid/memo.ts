import { memo } from 'react'

import Grid from './component'

export default memo(Grid)
