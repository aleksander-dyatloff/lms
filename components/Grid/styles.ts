import styled from 'styled-components'

const gridColumnsCount = 12

const Wrapper = styled.div`
  --gridSpacing: ${({ theme }) => theme.spacing * 3}px;

  display: flex;
  flex-wrap: wrap;
  margin-left: - var(--gridSpacing);
  margin-right: - var(--gridSpacing);

  & > * {
    width: 100%;
    max-width: 100%;
    padding-left: var(--gridSpacing);
    padding-right: var(--gridSpacing);
  }

  .col {
    flex: 1 0 0%;
  }
`
export default Wrapper
