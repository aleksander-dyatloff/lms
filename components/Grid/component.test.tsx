import 'jest-styled-components'

import { render } from '@testing-library/react'
import { Profiler } from 'react'

import Grid from './component'

describe('Grid component', () => {
  test('should match to snapshot', () => {
    const { container } = render(<Grid />)

    expect(container).toMatchSnapshot()
  })

  test('first render duration should match to specification', () => {
    render(
      <Profiler
        id='Grid'
        onRender={(_, __, renderDuration) => {
          expect(renderDuration).toBeLessThanOrEqual(1.5)
        }}
      >
        <Grid />
      </Profiler>,
    )
  })
})
