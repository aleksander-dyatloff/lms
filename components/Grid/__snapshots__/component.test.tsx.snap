// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component should match to snapshot 1`] = `
<div>
  <div
    class=""
  />
</div>
`;
